### DISCUSSION TITLE
Unable to upgrade PostgreSQL database

### DISCUSSION THREAD

**Discussion by DaniG2k** [Homebrew/discussions#4568] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.22
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9d01e1deafca35f30b709b5f6a031faaff9642ec
Last commit: 18 hours ago
Core tap JSON: 13 Jun 05:39 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I have a PostgreSQL 12.15 database. I'd like to upgrade it to a PostgreSQL 15 database.
I read that using `brew postgresql-upgrade-database` should perform the upgrade for me. However, this does not work, as I get the following message:

```
brew postgresql-upgrade-database
Warning: Formula postgresql was renamed to postgresql@14.
==> brew install postgresql@13
Error: No postgresql 13.* version installed!
```

I tried installing `postgresql@15` and running the command again to no avail.

Can someone assist me with upgrading a local database from PostgreSQL v12 to 15 please?
Thanks in advance

**Comment by gromgit** [comment]
See https://github.com/orgs/Homebrew/discussions/4451

**Comment by altivi** [comment]
@DaniG2k To perform the upgrade, you can follow these steps:

Uninstall the existing PostgreSQL formula:
`brew uninstall postgresql`

Install the PostgreSQL 14 formula:
`brew install postgresql@14`

Stop any running PostgreSQL services:
`brew services stop postgresql`

Upgrade the database to the new version using the postgresql-upgrade-database command:
`brew postgresql-upgrade-database`

Once the upgrade is complete, start the PostgreSQL service again:
`brew services start postgresql@14`

This will start the PostgreSQL 14 service.

Verify that the upgrade was successful by checking the PostgreSQL version:
`psql --version`

It should display the updated version, which should be 14.x.

By following these steps, you should be able to successfully upgrade your local PostgreSQL database from version 12 to version 14 using Homebrew.
